[FEAT] aws sagemaker python temlpate (#643)
pulumi/templates: aws-sagemaker-python/README.md

# AWS SageMaker + Hugging Face LLM Deployment

## Overview

A Pulumi IaC program written in Python to deploy a Hugging Face Language Model (LLM) on AWS SageMaker.

## Included:

- IAM roles
- SageMaker model endpoint
- CloudWatch alarms

## Prerequisites

> * Python 3.8+

Reviewer: I would change this to "a supported version of Python" and link to the Python docs _unless_ Python 3.8 is needed (and based on the "Python 3.8+" I'd guess it isn't). 3.8 is nearing EOL.

Author: changed to 3.9 in https://github.com/pulumi/templates/pull/643/commits/974781ff9aa0bd4fa4983b5603a6410e31119843

> # AWS SageMaker + Hugging Face LLM Deployment

Reviewer: Let's change this to "Amazon SageMaker" as well (and yes, I know the template name is `sagemaker-aws-python`...don't point out the inconsistencies)

Author: resolved in https://github.com/pulumi/templates/pull/643/commits/c4d9a4d56a45c5b2d638254a92ea89542a457945

## Overview

A Pulumi IaC program written in Python to deploy a Hugging Face Language Model (LLM) on AWS SageMaker.

## Included:

- IAM roles
- SageMaker model endpoint
- CloudWatch alarms

## Prerequisites

* Python 3.9+
* Pulumi
* AWS CLIv2 & valid credentials configured

## Quick Start

### Setup

1. Create a new directory & initialize a new project:

```bash
mkdir newProject && cd newProject
pulumi new aws-sagemaker-python
```

2. Deploy the stack:

```bash
pulumi up
```

> Note that Pulumi will provide the SageMaker endpoint name as an output.

### Test the SageMaker Endpoint

Use this rudimentary Python snippet to test the deployed SageMaker endpoint.

> 1. Activate the python `venv` locally

Reviewer: ```suggestion
1. Activate the Python `venv` locally
```

```bash
# On Linux & MacOS
source venv/bin/activate
```

2. Save the following as test.py:

> NOTE: change your `region_name` if using a different region than `us-east-1`

```python
import json, boto3, argparse

def main(endpoint_name):
    client = boto3.client('sagemaker-runtime', region_name='us-east-1')
    payload = json.dumps({"inputs": "In 3 words, name the biggest mountain on earth?"})
    response = client.invoke_endpoint(EndpointName=endpoint_name, ContentType="application/json", Body=payload)
    print("Response:", json.loads(response['Body'].read().decode()))

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("endpoint_name")
    main(parser.parse_args().endpoint_name)
```

2. Run the test:

> > replace the endpoint with your Pulumi outpu endpoint name:

Reviewer: ```suggestion
> replace the endpoint with the endpoint name in your Pulumi stack output:
```

```bash
python3 test.py Llama2Llm-endpoint-2bc72c2
```

> 

Author: resolved in https://github.com/pulumi/templates/pull/643/commits/c4d9a4d56a45c5b2d638254a92ea89542a457945

### Cleanup

> To destroy the Pulumi stack and all of it's resources:

Reviewer: ```suggestion
To destroy the Pulumi stack and all of its resources:
```
